File: src/cpp/argv0_test.cpp
#include <cstring>
#include <iostream>
#include <string>
#include <vector>

int main(int argc, char *argv[]) {
  if (argc < 1) {
    std::cerr << "Error: argc is less than 1" << std::endl;
    return 1;
  }

  if (strcmp(argv[0], "/main.wasm") != 0) {
    std::cerr << "Error: argv[0] is \"" << argv[0]
              << "\", expected \"/main.wasm\"" << std::endl;
    return 1;
  }

  std::cout << "Success: argv[0] is \"/main.wasm\"" << std::endl;
  return 0;
}

File: src/go/argv0_test.go
package main

import (
	"fmt"
	"os"
)

func main() {
	if len(os.Args) < 1 {
		fmt.Printf("Error: os.Args is empty\n")
		os.Exit(1)
	}
	if os.Args[0] != "/main.wasm" {
		fmt.Printf("Error: os.Args[0] is %q, expected \"/main.wasm\"\n", os.Args[0])
		os.Exit(1)
	}
	fmt.Printf("Success: os.Args[0] is \"/main.wasm\"\n")
}
